packages/syft/src/syft/core/tensor/autodp/phi_tensor.py

```diff
@@ -1087,11 +1088,70 @@ def copy(self, order: Optional[str] = "K") -> PhiTensor:
             data_subjects=self.data_subjects.copy(),
         )
 
-    def all(self) -> bool:
-        return self.child.all()
+    def any(
+        self,
+        axis: Optional[Union[int, Tuple[int, ...]]] = None,
+        keepdims: Optional[bool] = False,
+        where: Optional[ArrayLike] = None,
+    ) -> PhiTensor:
+        # TODO: properly define data subjects and
+        # figure out if it is not a privacy violation to return bool
+        if where is None:
+            out_child = np.array(self.child.any(axis=axis, keepdims=keepdims))
+        else:
+            out_child = np.array(
+                self.child.any(axis=axis, keepdims=keepdims, where=where)
+            )
+
+        new_data_subjects = DataSubjectArray.from_objs(np.empty(out_child.shape))
```

Comment:
I'm curious as to why we wouldn't be sourcing data subjects from "self" here in some way. Surely "self" could be referring to a particular data subject, in which case the new_data_subjects variable would need to fetch from it, yes?

Reply:
(ditto for other methods)

Reply:
the main issue was that I didn't know to whom to "assign" the new values from the data subjects as the shape of the result could be very different from the shape of the data subjects. I guess we could reduce over the right axis with the add operation on the data subjects, At the same time, I didn't know if we need to implement any and all like that or if we consider the release of the result themselves to be a privacy violation. The main problem is that there are other sections of the code where we use all and any without the privacy wrapper, I guess maybe we can change it only to use them on the child values

Reply:
fixed